server/src/main/java/org/opensearch/index/compositeindex/datacube/ReadDimension.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube;
+
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Composite index merge dimension class
+ *
+ * @opensearch.experimental
+ */
+public class ReadDimension implements Dimension {
```

Comment:
This is ReadDimension. It is used when we need to reconstruct StarTreeField from the file formats. Since we don't need all of the information, and only the names, it helps us with an easy interface to construct start tree field during searches / merges.

Reply:
Please add the same to javadoc.